src/object_test.ts

```diff
@@ -1921,5 +1921,31 @@ describe('KubernetesObject', () => {
             expect(thrown).to.be.true;
             scope.done();
         });
+
+        it('should throw error if no apiVersion', async () => {
+            let thrown = false;
+            try {
+                await (client.list as any)(undefined, undefined);
+                expect.fail('should have thrown an error');
+            } catch (e) {
+                thrown = true;
+                expect(e.message).to.contain(
+                    'Required parameter apiVersion was null or undefined when calling ',
+                );
+            }
+            expect(thrown).to.be.true;
+        });
+
+        it('should throw error if no kind', async () => {
+            let thrown = false;
+            try {
+                await (client.list as any)('', undefined);
+                expect.fail('should have thrown an error');
+            } catch (e) {
+                thrown = true;
+                expect(e.message).to.contain('Required parameter kind was null or undefined when calling ');
+            }
+            expect(thrown).to.be.true;
+        });
```

Comment:
Can you add a test that validates that you correctly construct the URL for the GET then I think this is ready to merge.

Reply:
Added tests for namespaced and non namespaced list requests.
And also added support for the continue feature